packages/plugin-ext/src/plugin/plugin-manager.ts

```diff
@@ -261,38 +259,49 @@ export class PluginManagerExtImpl implements PluginManagerExt, PluginManager {
         let loading = this.loadedPlugins.get(plugin.model.id);
         if (!loading) {
             loading = (async () => {
-                if (plugin.rawModel.extensionDependencies) {
-                    for (const dependencyId of plugin.rawModel.extensionDependencies) {
-                        const dependency = this.registry.get(dependencyId.toLowerCase());
-                        const id = plugin.model.displayName || plugin.model.id;
-                        if (dependency) {
-                            const depId = dependency.model.displayName || dependency.model.id;
-                            const loadedSuccessfully = await this.loadPlugin(dependency, configStorage, visited);
-                            if (!loadedSuccessfully) {
-                                const message = `Cannot activate extension '${id}' because it depends on extension '${depId}', which failed to activate.`;
+                const progressId = await this.notificationMain.$startProgress({
+                    title: `Activating ${plugin.model.displayName || plugin.model.name}`,
+                    location: 'window'
+                });
+                try {
+                    if (plugin.rawModel.extensionDependencies) {
+                        for (const dependencyId of plugin.rawModel.extensionDependencies) {
+                            const dependency = this.registry.get(dependencyId.toLowerCase());
+                            const id = plugin.model.displayName || plugin.model.id;
+                            if (dependency) {
+                                const depId = dependency.model.displayName || dependency.model.id;
+                                const loadedSuccessfully = await this.loadPlugin(dependency, configStorage, visited);
```

Comment:
TODO: We should not pass `configStorage`, but use `this.configStorage` when actual load is happening, otherwise it can race with `$updateStoragePath`